### ISSUE TITLE
Problems Compiling Kratos

### ISSUE THREAD

**Issue by xaviertort** [KratosMultiphysics/Kratos#11426]
Dear all,

I'm trying to compile Kratos, inside poro/test-pullout-new (PoromechanicsApplication) on my PC but I'm having problems.

The environment in which I'm trying to compile Kratos is the following:

- Windows 10, Version 1709, compilation 16299.2166
- Visual Studio 2019
- CMake 3.27.0
- Python 3.9.7
- Boost 1.78.0

I tried to compile Kratos on my laptop by using the exactly same environment expect for the operative system (my laptop uses Windows 11), and it worked well, hence, I have the latest version of the software and I can use it. So, this is why I do not understand where's the error (maybe on the operative system??).

I attache some screenshots to show you the errors I get every time I try to compile. The first two screenshots (1 and 2) are two different warnings I get at the beggining about CMake. Even though I get these messages, it seems that the compilation is working well and the procedure seems normal.

But then, when it feels like the compilation is about to finish, I get the error shown in third screenshot (3) lots of times, and the "compilation" finishes with the error seen in screenshot 4.

If anyone knows what is the problem or can help me fixing it I will really appreciate because I need it working as soon as possible.

Thank you in advance.

![1](https://github.com/KratosMultiphysics/Kratos/assets/88047375/428abe9d-442b-4f37-8081-235da930d9a5)
![2](https://github.com/KratosMultiphysics/Kratos/assets/88047375/6a3a11b7-66e6-48bb-aefd-9df4b8e7e944)
![3](https://github.com/KratosMultiphysics/Kratos/assets/88047375/9c5662e4-4a06-443e-87f1-ee422a45fbe0)
![4](https://github.com/KratosMultiphysics/Kratos/assets/88047375/3e75a4f5-c76d-487b-8907-121a82409f99)

**Comment by ipouplana** [comment]
Hi Xavi, can you copy the configuration file you are using to compile? Also, I think it would be helpful to see the first lines of the compilation output (from the beginning up to the list of applications being compiled).

**Comment by roigcarlo** [comment]
Hi @xaviertort ,

Can you post the configure.bat and the full output? (if possible in text format)

**Comment by xaviertort** [comment]
> Open the command window with admin rights

Hi Alejandro,

I already done in admin rights (see screenshot 4). This is why I don't understand where is the problem.

Even though, thank you for your advice

**Comment by xaviertort** [comment]
> Hi @xaviertort ,
> 
> Can you post the configure.bat and the full output? (if possible in text format)


Hi @roigcarlo,

I post three different files. The first one (kratos_poromechanics_debug.txt) is the one I'm using to try to compile Kratos. The second one (cmd_output1.txt) is the response I'm getting at the beggining when I try to run the .bat file.

Finally, the third file (cmd_output_final_error.txt) is the error message I get at the end, which corresponds to the message I added via screenshot on the first Issue.

The output is attached in two different parts because the output is like cutted, so I stopped manually in order to obtain the first lines of this output.

Hope this will help you.

Thank you in advance.


[kratos_poromechanics_debug.txt](https://github.com/KratosMultiphysics/Kratos/files/12168982/kratos_poromechanics_debug.txt)
[cmd_output1.txt](https://github.com/KratosMultiphysics/Kratos/files/12168983/cmd_output1.txt)
[cmd_output_final_error.txt](https://github.com/KratosMultiphysics/Kratos/files/12168986/cmd_output_final_error.txt)

**Comment by roigcarlo** [comment]
Hi, 

I believe the problem are the spaces in your username. Try to put the paths were your username appears quoted:

```
rem Set basic configuration
set KRATOS_BUILD_TYPE=FullDebug
if not defined BOOST_ROOT set BOOST_ROOT="C:\Users\Xavi Tort\Repositoris"
if not defined PYTHON_EXECUTABLE set PYTHON_EXECUTABLE="C:\Users\Xavi Tort\AppData\Local\Programs\Python\Python39\python.exe"
```

**Comment by xaviertort** [comment]
Hi @roigcarlo @matekelemen,

I tried all the alternatives proposed by you and I am still getting error messages, but different that the originals this time. I attache the output on this comment. I tried to understand them but I don't have the corresponding knowledge. 

I am working on Campus Nord, building C1, room C6 (only on the mornings), in case you would like to come and try anything.

Thank you

[cmd_output_270723.txt](https://github.com/KratosMultiphysics/Kratos/files/12181968/cmd_output_270723.txt)

**Comment by roigcarlo** [comment]
are you suing cmake 3.27 or greater?

**Comment by xaviertort** [comment]
> are you suing cmake 3.27 or greater?

Yes, 3.27.0, as in my laptop. Would it be better an older version?

Thank you again and sorry for all the inconvenience😅

**Comment by roigcarlo** [comment]
Yep seems that part of the problem is the cmake version. Could you try to see if it works with 3.26. I will try to fix it in the meantime.

**Comment by xaviertort** [comment]
> Yep seems that part of the problem is the cmake version. Could you try to see if it works with 3.26. I will try to fix it in the meantime.

Hi Carlos,

On the one hand, I tried to compile with cmake 3.26.5 and I'm obtaining the same error. On the other hand, I tried to compile with the changes you proposed on the pull request but it doesn't work either.

I attache the output in case it can help you.


[cmd_output_280723.txt](https://github.com/KratosMultiphysics/Kratos/files/12191965/cmd_output_280723.txt)
